### ISSUE TITLE
Cannot create a new conversation 

### ISSUE THREAD

**Issue by AbdullahQureshi1080** [twilio/twilio-conversations-demo-react#144]
I was trying to test the demo out, after having done the initial steps and setting up the serverless functions. I can log in and build the app successfully. However, I cannot create a new conversation. 

`Error: Cannot read properties of undefined (reading 'then')
`

```
  "react": "17.0.2",
  "@twilio/conversations": "2.4.1",
```

It's this function

```
  async createConversation(options) {    
        await this._ensureReady;
        options = options || {};
        return this._conversationsPromise.then((conversationsEntity) => conversationsEntity.addConversation(options));
    }
```

_Note: I am using the serverless functions for the backend._ 

Node: v18

Use the steps [here](https://github.com/twilio/twilio-conversations-demo-react#next-steps) to set up the service. 

<img width="1440" alt="Screenshot 2023-12-06 at 10 51 07 AM" src="https://github.com/twilio/twilio-conversations-demo-react/assets/55753562/0dce8d11-4b27-4322-9d07-ce0f6f2dec91">

I have gone through the other issues, not that helpful. Anyone who's been able to get throught this, I'd appreciate the support.

**Comment by AbdullahQureshi1080** [comment]
[makzpatel](https://github.com/makzpatel), [camr1993](https://github.com/camr1993), [esquive](https://github.com/lesquive) Any help here. I have the project cloned locally and running. If it'll help I can set up a repo and share that.

**Comment by jmozzart** [comment]
Check the variables in the functions. I had the same problem before and found out that I mispelled the variables.

**Comment by jmozzart** [comment]
Check both.

**Comment by AbdullahQureshi1080** [comment]
In the serverless function

```
let token = new AccessToken(
      context.ACCOUNT_SID,
      context.TWILIO_API_KEY_SID,
      context.TWILIO_API_KEY_SECRET, {
        identity: event.identity,
        ttl: 3600
      });

    let grant = new AccessToken.ChatGrant({ serviceSid: context.SERVICE_SID });
```


Variables (ENV)


<img width="474" alt="Screenshot 2023-12-07 at 12 49 09 PM" src="https://github.com/twilio/twilio-conversations-demo-react/assets/55753562/741c4bbc-36b0-4f89-9aab-a630b77562d3">

Add my Twilio Credentials (ACCOUNT_SID) and (AUTH_TOKEN) to ENV -> This is also checked to be true. 

So this is in the serverless config!

Do I have to add these variables in the demo project on the front end as well?

**Comment by lesquive** [comment]
@berkus maybe we can change `Error: Cannot read properties of undefined (reading 'then')` to a different message so it doesn't confuse anyone.

**Comment by berkus** [comment]
@lesquive sure, we have almost non-existing error handling in the app, there are many places where we could improve error messages and handle them gentler.

**Comment by GomezFabio** [comment]
I had the same problem and this is how I have solved it.
I was using wrong credentials for **TWILIO_API_KEY_SID** and **TWILIO_API_KEY_SECRET** variables.
You must create a new API Key and get the right variables from it.
Check it at https://console.twilio.com/us1/account/keys-credentials/api-keys and use [Create API key] button.

**Comment by AbdullahQureshi1080** [comment]
@GomezFabio Thanks. 

I already did this. I am able to use auth but cannot create room/channel for testing the whole thing out. 

And I was not able to find anything in the demo, example or the docs on how to trace my steps for the setup. 

Currently putting twilio on hold, exploring other options.

**Comment by Fabryz** [comment]
I've just deployed to Vercel, followed the guide in the github readme of the demo, I also get this error when creating a new conversation:

```
Cannot read properties of undefined (reading 'then')
```

Any news on this? @AbdullahQureshi1080

**Comment by AbdullahQureshi1080** [comment]
No update, we moved to another sdk as their solution did not fulfill our
use case.

**Comment by Fabryz** [comment]
> > My app is deployed on Vercel, I saw "offline" on the top bar under my username.
> 
> Did you set up the token generator?
I've setup the two TWILIO API KEYS

Could you please tell me what I'd need to do in particular?

**Comment by berkus** [comment]
> Could you please tell me what I'd need to do in particular?

You need to deploy a function as descibed in the README that will generate and provide you with access tokens. When deploying to vercel it shall ask you about the token generator url - you need to provide that function's url there.

**Comment by alorian** [comment]
@GomezFabio gave a right answer. Thank you. My problem was in the existing API key which is actually not an API key at all. 

https://console.twilio.com/us1/account/keys-credentials/api-keys
This page has two sections. 
1. API keys 
2. Auth Tokens

There was a live auth token which I already used before this demo application and this auth token didn't work.

Now I've created a new **API key**. I've updated my env variables in the cloud function and everything works now. The status is set to "online" after the initial sign in. 

@Fabryz FYI
